proposals/pr-contrib-guide.md

```diff
@@ -0,0 +1,531 @@
+issue captures information necessary to make the contribution, such as:
+
+1. A description of the requested change.
+1. The number of users in the community that request the change, measured
+   through GitHub :+1: reactions.
+1. Design discussions, including use cases, edge cases, and alternative
+   solutions.
+
+Contributions made without a GitHub issue will be subject to closure by the
+maintainers.
+
+Maintainers will regularly review open GitHub issues to determine if they are
+ready for community contribution. The decision to label an issue as ready for
+community contribution is ultimately up to the maintainers. Depending on the
+issue type (bug or feature request) and area (code or documentation), the
+maintainers will use a set of criteria to decide if the issue is ready for
+contribution. This may include:
+
+1. The change is feasible for a community member to contribute. See [the
+   appendix](#changes-not-available-for-contribution) for details.
+1. Feature requests must have at least 10 :+1: reactions.
+1. The maintainers agree with the implementation plan.
+
+If the criteria are not satisfied, the maintainers will comment on the GitHub
+issue to request more information and to update the current status or next
+steps. Any design discussions (e.g., public interface design, edge cases,
+backwards compatibility concerns) must be resolved prior to making a
+contribution so that feedback during the review process focuses on the
+implementation of the agreed upon design.
+
+Issues that do not receive enough :+1: reactions and do not have any activty for
+a year are considered stale. The maintainers will make a request for comment
+from the community. Without further interaction, the issue will be closed.
+
+### Contribution-ready
+
+If the criteria for ready for contribution are satisfied, the issue will be
+labelled to identify it as ready to contribute. This new stage clearly
+identifies which issues the maintainers have determined are ready for
+contribution and makes them visible for users can discover them. Any issue that
+has been marked as available is open for a community contribution through a
+GitHub pull request.
+
+### Implementation
+
+We will improve the implementation stage with specific instructions and criteria
+for completing an implementation. A user can select an issue from the list of
+contribution ready issues and implement their change. The contributor should use
+the CONTRIBUTING guide as a reference for the criteria to be met before a
+contribution will be reviewed by the maintainers. This criteria includes passing
+validation checks, passing new and existing tests, code linting, and style
+checks. The contributor can also communicate with the maintainers on the issue
+with questions that arise. 
+
+When the contributor has completed their implementation, they should open a pull
+request. The contribution is then given an initial inspection by the maintainers
```

Comment:
We should hoist this sentence to the beginning of the section. In general for all of these stages sections as the beginning, we should clearly state:

1. How do you get to this stage
2. What does this stage mean

Right now these two points are buried, and while it sort of intuitive that implementation stage would be when a PR gets submitted and represents that a contribution is being worked on. We should explicitly call it out from the start so that nobody makes the incorrect assumption about what it represents before getting too deep into the details (e.g. review prerequisites, contributing guide). I also think we can make these changes in some of the following sections as well skimming through them.